packages/middleware/test/build.test.ts

```diff
@@ -54,4 +54,29 @@ describe('build()', () => {
       (unhandledResponse.response as Response).headers.get('x-middleware-next')
     ).toEqual('1');
   });
+  it('should build simple middleware with env vars', async () => {
+    const expectedEnvVar = 'expected-env-var';
+    const fixture = join(__dirname, 'fixtures/env');
+    process.env.ENV_VAR_SHOULD_BE_DEFINED = expectedEnvVar;
+    await build({
+      workPath: fixture,
+    });
+    // env var should be inlined in the output
+    delete process.env.ENV_VAR_SHOULD_BE_DEFINED;
+
+    const outputFile = join(fixture, '.output/server/pages/_middleware.js');
+    expect(await fsp.stat(outputFile)).toBeTruthy();
+
+    require(outputFile);
```

Comment:
Rather than polluting the global scope, can we use the `vm` module instead?

Reply:
I copied what has already been done in the other test here
https://github.com/vercel/vercel/blob/c42efe6ecdebd914dd961f89f7d9acafb4dad500/packages/middleware/test/build.test.ts#L34

I think it would be more appropriate to migrate to `vm` in another PR if you want

Reply:
Do you agree?

Reply:
Ya sorry, I see other tests doing that as well now. So disregard my comment here in this case.